op-service/clock/advancing.go

```diff
@@ -0,0 +1,71 @@
+package clock
+
+import (
+	"sync/atomic"
+	"time"
+)
+
+type AdvancingClock struct {
+	*DeterministicClock
+	systemTime   Clock
+	ticker       Ticker
+	advanceEvery time.Duration
+	quit         chan interface{}
+	running      atomic.Bool
+
+	lastTick time.Time
+}
+
+// NewAdvancingClock creates a clock that, when started, advances at the same rate as the system clock but
+// can also be advanced arbitrary amounts using the AdvanceBy method.
```

Comment:
I think this is referencing the `AdvanceTime` method in `DeterministicClock`
```suggestion
// can also be advanced arbitrary amounts using the AdvanceTime method.
```